src/nvim/func_attr.h

```diff
@@ -212,6 +212,8 @@
 #if defined(DEFINE_FUNC_ATTRIBUTES) || defined(DEFINE_EMPTY_ATTRIBUTES)
 /// Fast (non-deferred) API function.
 # define FUNC_API_FAST
+/// Return value needs to be freed
```

Comment:
```suggestion
/// Return value must be freed. For non-allocating return, add an `Arena` parameter to the function signature.
```

Reply:
The mere _presence_ of an arena parameter does not change _any_ behavior anymore, so this would be quite misleading, as an API function can just as well return Object values which has a longer, external lifetime (like channel info is a good example). 

The point of an arena is just that the arena can be _used_ to allocate values which doesn't need to be xfree:d, which can be (part of) the return value, or other stuff besides that which just needs any temporary memory, which is why several `void` returning functions also use it.